### ISSUE TITLE
Add mechanism for turning on daily output in test suite

### ISSUE THREAD

**Issue by mnlevy1981** [ESCOMP/MOM_interface#167]
The [`defaultio`](https://github.com/ESCOMP/CESM/tree/cesm2_3_beta17/cime_config/testmods_dirs/allactive/defaultio) test in CESM writes daily history files for CAM, CICE, CISM, CLM, MOSART, POP, and RTM. It would be nice to have an ability to adjust the MOM `diag_table` for this test.

I proposed to @jedwards4b that we could add a `DAILY_OUTPUT` variable to `env_test.xml` (default=`FALSE`), and when `DAILY_OUTPUT=TRUE` ParamGen would modify the diag table accordingly, but I am open to other methods of getting daily MOM output into the test suite.

**Comment by alperaltuntas** [comment]
@mnlevy1981 Is this issue still relevant?

**Comment by jedwards4b** [comment]
@alperaltuntas yes I think that it is.  Case  /glade/derecho/scratch/jedwards/PFS.ne30pg3_t232.BLT1850.derecho_intel.allactive-defaultio.20241226_071515_19nl23 has daily output for all components except MOM.

**Comment by jedwards4b** [comment]
I'm not sure that there is a need to do this with an additional XML variable - what exactly are the changes that need to happen in diag_table and why couldn't it be done with a testmod as in other components?
